client/src/main/java/io/pravega/client/stream/Stream.java

```diff
@@ -24,6 +24,14 @@
  * Scalable (The rate of events in a stream can greatly exceed the capacity of any single host)
  */
 public interface Stream {
+    enum State {
```

Comment:
I'm not sure this belongs here. This is a public interface exposed to our customers. It is also not used anywhere in this interface, so it would be confusing.

Reply:
this has been moved out to impl/StreamState